### PR TITLE
[vsphere] define mirror registry to be used for selected 4.6 e2e tests

### PR DESCRIPTION
**Pull request by rvanderp3**
TL;DR:
- Allow definition of environment variable which openshift-tests recognizes as a source for images for some tests that pull directly from docker.io.
- Configure vSphere 4.6 jobs to define that variable to point to a mirror present in VMC.

Details:
This PR is related to a PR in [origin](https://github.com/openshift/origin/pull/26729) which introduces an environment variable which is used by targeted tests to determine where a test should pull it's related image.  The following changes are intended to address docker.io pull rate issues on some platforms:
1. A pull-through cache which acts as a registry mirror is deployed
2. A mirror configuration which defines the registry mirror as a mirror of docker.io.  This is deployed as a machine config(see [vSphere example](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/vsphere/registry/ipi-install-vsphere-registry-commands.sh#L34-L52)) as `ImageContentSourcePolicy` only allows a repository digest mirror to be defined.  
3. Updates to `openshift-tests` to enable specific tests which directly pull images from the context of the pod where `openshift-tests` is running to pull from the mirror.

1 and 2 are presently implemented for vSphere jobs and have proven successful for serial 4.6 jobs.

### CODE REVIEW

**Comment by rvanderp3** [comment]
4.6 vSphere parallel jobs passed!

File: ci-operator/step-registry/openshift/e2e/vsphere/upi/release46/openshift-e2e-vsphere-upi-release46-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+workflow:
+  as: openshift-e2e-vsphere-upi-release46
```
Comment (vrutkovs) [inline ci-operator/step-registry/openshift/e2e/vsphere/upi/release46/openshift-e2e-vsphere-upi-release46-workflow.yaml:2]: This workflow is incredibly useful for OKD jobs, which uses a lot of docker.io images. Lets rename this to `openshift-e2e-vsphere-upi-pullthrough` or similar?

Comment (rvanderp3) [reply ci-operator/step-registry/openshift/e2e/vsphere/upi/release46/openshift-e2e-vsphere-upi-release46-workflow.yaml:2]: The reason for specifically calling out release-4.6 in the name is due to the fact that this configuration is not necessary in 4.7 and later as the targeted tests in openshift-tests already pull from other sources.  The bulk of the tests which failed due to rate limiting are fixed by configuring a mirror in crio.

Comment (vrutkovs) [reply ci-operator/step-registry/openshift/e2e/vsphere/upi/release46/openshift-e2e-vsphere-upi-release46-workflow.yaml:2]: Ah, I see, okay
